packages/editor/src/lib/editor/shapes/ShapeUtil.ts

```diff
@@ -373,6 +373,10 @@ export abstract class ShapeUtil<Shape extends TLUnknownShape = TLUnknownShape> {
 		return {}
 	}
 
+	getText(_shape: Shape): string | undefined {
```

Comment:
since we're thinking of this new hook into the API, i think we should think of the general case where there is multiple text snippets, potentially, within a shape. so, i believe this should actually return:
`string[] | undefined` (and then the shape utils below should also return arrays).

_potentially_, that might mean it also returns more information? like id or position or something but i don't think we should get too complicated / over think this.

but i do think we should at least account for multiple textfields.

Reply:
I kinda like the lean approach. Until we see a use case I'd rather use the most simple api, which works for the current system. Stuff gets complicated enough already 😄

Reply:
I feel ya on the lean approach. I just worry about the upgrade path here down the line. It feels like, yes, currently we only have one text field per shape. But this then doesn't become a usable API for other folks that have custom shapes (non-tldraw) that have multiple text fields. (there's feature requests that are asking for labels on both sides of an arrow for example). 

A different example would be how we have various checks in our geometry code to see if `isLabel` is true, because we don't want to bake in an assumption that there's only ever one label in the geometry. It keeps the optionality for multiple labels to exist open.

Reply:
I'm happy to change the api to return an array of strings instead of just a single one. As far as going further than that I don't feel confident in getting the API right without seeing more examples of how it could be used.

I don't think updating this in the future is that big of a problem. It's a fairly contained API and even if we do introduce breaking changes they should be quite limited in scope.

Reply:
Happy to let you take over if you understand the possible use cases better.

Reply:
Yeah, some examples of how users could use it:
- tables: https://github.com/tldraw/tldraw/issues/1780
- arrows with multiple labels: https://github.com/tldraw/tldraw/issues/2141

And, again, we have to think about how we might have more primitive shapes (square, circle) on our board but our downstream users will have more complex shapes. So, another example could be, say, a slideshow creating tool. And there's a shape that let's you create title slides with Title, Subtitle, and a Date or something. Maybe they could compose that with separate text shapes individually each time but I could image that being annoying and having a combined Slide shape that encompasses multiple text labels like that.

But yeah, the table example above also shows a pretty good use case!

Reply:
It's still hypothetical. We don't have tables. Would you then expect the API to return an array of arrays for the tables?  Or just a flat array? Would it have some additional info associated with it (like positions)?
That's what I'm talking about when I say I don't understand the use case enough to be confident in coming up with an api.

Like I said I'm happy to change it from a single string to an array of strings if you feel that's better (I don't for now). But if you'd like more than that then you can either take this on or I can just close the PR if you feel it's not useful as is.

Reply:
Hah, yeah, I know we don't have tables, I don't think we ever will in tldraw formally. But _other_ people will add tables to their usage of tldraw, and then they'll want to maybe use the search functionality.

> Would it have some additional info associated with it (like positions)?

Yes, I saw that above in my first comment: "potentially, that might mean it also returns more information? like id or position or something but i don't think we should get too complicated / over think this. but i do think we should at least account for multiple textfields."

Maybe we should talk offline at this point?  :)

Reply:
Talked offline and we'll go ahead land as is!
We'll naturally revisit all of this work anyway because of things like rich text, community feedback, and other feature requests.